ide/projectui/src/org/netbeans/modules/project/ui/OpenProjectList.java

```diff
@@ -706,6 +707,12 @@ public void open(Project[] projects, boolean openSubprojects, ProgressHandle han
                 } else {
                     LOGGER.log(Level.WARNING, "Project in {0} disappeared", p.getProjectDirectory());
                 }
+                if (prime) {
+                    ActionProvider ap = p2.getLookup().lookup(ActionProvider.class);
+                    if (ap != null && ap.isActionEnabled(ActionProvider.COMMAND_PRIME, p2.getLookup())) {
+                        ap.invokeAction(ActionProvider.COMMAND_PRIME, p2.getLookup());
```

Comment:
The queries executed below query containment. It could be better to wait for PRIME action to finish, so e.g. gradle has a chance to load its full model and discover the project's content at all.

Reply:
Shouldn't load of a full model and set of trust happen as a result of `COMMAND_PRIME`? I believe it happens. As such I wanted to run the priming as soon as possible.

Reply:
Understood, but the Prime action may fork the execution into its own RP; it does not necessarily complete synchronously within `invokeAction` scope.

Reply:
Is a772b3f OK as a fix?